src/routes/safe/components/Apps/components/SecurityFeedbackModal/Slider.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useState, useRef, useEffect } from 'react'
+import styled from 'styled-components'
+import { Box } from '@material-ui/core'
+import { Button } from '@gnosis.pm/safe-react-components'
+
+type SliderProps = {
+  onCancel: () => void
+  onSlideChange: (slideIndex: number) => void
+  onComplete: () => void
+}
+type SliderState = {
+  translate: number
+  activeSlide: number
+  transition: number
+  renderedSlides: React.ReactElement[]
+}
+
+const SLIDER_TIMEOUT = 500
+
+const Slider: React.FC<SliderProps> = ({ onCancel, onSlideChange, onComplete, children }) => {
+  const allSlides = React.Children.toArray(children).filter(Boolean) as React.ReactElement[]
+  const stateRef = useRef<SliderState>({
+    translate: 1,
+    activeSlide: 0,
+    transition: 0.45,
+    renderedSlides: [],
+  })
+  const firstSlide = allSlides?.[0]
+  const secondSlide = allSlides?.[1]
+  const lastSlide = allSlides?.[allSlides?.length - 1]
+  const [disabledBtn, setDisabledBtn] = useState(false)
+  const [state, setState] = useState<SliderState>({
+    translate: 1,
+    activeSlide: 0,
+    transition: 0.45,
+    renderedSlides: [lastSlide, firstSlide, secondSlide],
+  })
+
+  stateRef.current = state
+
+  useEffect(() => {
+    const smooth = (e: TransitionEvent) => {
+      if ((e.target as HTMLElement).className.includes('inner')) {
+        smoothTransition()
+      }
+    }
+
+    document.addEventListener('transitionend', smooth)
+
+    return () => {
+      document.removeEventListener('transitionend', smooth)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```

Comment:
I think that we should add  `smoothTransition` to this array of dependencies and remove the 
```
// eslint-disable-next-line react-hooks/exhaustive-deps
```
I think that this should work:

```
  const smoothTransition = useCallback(() => {
  
      // smoothTransition code
      
    })
  }, [allSlides, firstSlide, lastSlide, secondSlide])

  useEffect(() => {
  
      // useEffect code
      
  }, [smoothTransition])
```

Reply:
mm but with this change the cleanup fn is triggered a lot of of times...

```
     return () => {
      document.removeEventListener('transitionend', smooth)
    }
    ```

Reply:
Updated!